tests/test_sql_insert_writer.py

```diff
@@ -0,0 +1,295 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""Tests for `sql_insert_writer` package."""
+
+import os
+import subprocess
+import sqlite3
+import tempfile
+
+import pytest
+import pytest_postgresql
+
+from click import BadOptionUsage
+from click.testing import CliRunner
+
+from sql_insert_writer import cli, sql_insert_writer
+
+try:
+    subprocess.check_output('command -v pg_ctl', shell=True)
+    PG_CTL_MISSING = False  # sorry for the double-negative, but it's convenient later
+except subprocess.CalledProcessError:
+    PG_CTL_MISSING = True
+
+
+def dsn_to_url(engine, dsn):
+    """
+    Converts a DSN to a SQLAlchemy-style database URL
+
+    pytest_postgresql connection only available in DSN form, like
+    'dbname=tests user=postgres host=127.0.0.1 port=41663'
+    """
+    params = dict(s.split('=') for s in dsn.split())
+    return '{engine}://{user}@{host}:{port}/{dbname}'.format(engine=engine,
+                                                             **params)
+
+
+TABLE_DEFINITIONS = [
+    'CREATE TABLE tab1 (col1 serial primary key, col2 text, col3 text, col4 text)',
+    'CREATE TABLE tab2 (col1 serial primary key, col3 text, col4 text)',
+    'CREATE TABLE tab3 (col1 serial primary key, col2 text, col4 text)',
+    'CREATE TABLE tab4 (col1 serial primary key, col2 text, col3 text)',
+    'CREATE TABLE tab5 (col1 serial primary key, datecol1 date, intcol1 integer, col2 text)',
+    'CREATE TABLE tab5_all_text (col1 serial primary key, datecol1 text, intcol1 text, col2 text)',
+]
+
+# pytest-postgresql works locally, but it contains an unshakeable
+# assumption that it needs to locally manage postgres instances with
+# a local installation of `pg_ctl`, which is incompatible with CircleCI.
+
+
+@pytest.mark.skipif(PG_CTL_MISSING, reason='PostgreSQL not installed locally')
+@pytest.fixture
+def pg_url(postgresql):
+    cur = postgresql.cursor()
+    for table_definition in TABLE_DEFINITIONS:
+        cur.execute(table_definition)
+    postgresql.commit()
+    db_url = dsn_to_url('postgresql', postgresql.dsn)
+    return db_url
+
+
+@pytest.fixture
+def sqlite_url(request):
+    sqlite_file = tempfile.NamedTemporaryFile(delete=False)
+
+    def teardown():
+        os.unlink(sqlite_file.name)
+
+    request.addfinalizer(teardown)
+
+    conn = sqlite3.connect(sqlite_file.name)
+    cur = conn.cursor()
+    for table_definition in TABLE_DEFINITIONS:
+        cur.execute(table_definition)
+    conn.commit()
+    return 'sqlite:///' + sqlite_file.name
+
+# @pytest.fixture(params=[sqlite_url, pg_url])
+# def all_db_with_tables(request, postgresql):
+#     return request.param(request=request, postgresql=postgresql)
+#     # Fails with pg_url() got an unexpected keyword argument 'request'
+#     # or AttributeError: 'function' object has no attribute '_instantiate_plugins'
```

Comment:
Should this commented-out code be removed?

Reply:
Yes indeed.